src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIProxy.cs

```diff
@@ -675,7 +675,7 @@ private void ReportSNIError(SNIProviders provider)
         private bool InferNamedPipesInformation()
         {
             // If we have a datasource beginning with a pipe or we have already determined that the protocol is Named Pipe
-            if (_dataSourceAfterTrimmingProtocol.StartsWith(PipeBeginning) || _connectionProtocol == Protocol.NP)
+            if (_dataSourceAfterTrimmingProtocol.StartsWith(PipeBeginning, StringComparison.OrdinalIgnoreCase) || _connectionProtocol == Protocol.NP)
```

Comment:
No need to ignore case. We didn't before and are just comparing to a specific character.
```suggestion
            if (_dataSourceAfterTrimmingProtocol.StartsWith(PipeBeginning, StringComparison.Ordinal) || _connectionProtocol == Protocol.NP)
```

Reply:
I assumed that _was_ the fix. There are no other changes in the file. Can we get an explanation of how this fixes the bug?

Reply:
The StringComparison.Ordinal is now used instead.  The issue was that using {th-TH} culture info, string.StartsWith always returns true when compared with @"\\" even when the string does not. So, a StringComparison option is used.

Reply:
That's a very strange behaviour. Did you reach out to someone on the runtime team to determine if that's correct? if not it might be a good idea, possibly open an issue in dotnet/runtime.

Reply:
@Wraith2 the issue happens on netcoreapp3.1 and is not happening on greater versions. I think it is not going to be fixed as netcoreapp3.1 is not supported anymore, but SqlClient v3.x is a long term supported version and we have to provide a fix for that. According to [this documentation](https://learn.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings#ordinal-string-operations)

> Ordinal comparisons are string comparisons in which each byte of each string is compared without linguistic interpretation

Also, look at [this example and notes
](https://learn.microsoft.com/en-us/dotnet/api/system.stringcomparison?view=netcore-3.1#examples)
I think this is a safe approach for string comparison in our driver. What do you think considering all of the above?

Reply:
If it's 3.1 only it was probably fixed in the later versions since they moved to using icu on all platforms. So as as a fix for an LTS it's fine with me.